apps/web/src/components/PublishMenu.tsx

```diff
@@ -10,6 +10,22 @@ import { Popover, PopoverContent, PopoverTrigger } from "./ui/popover";
 import { Button, buttonVariants } from "./ui/button";
 import { cn } from "../lib/utils";
 import { useToast } from "./ui/use-toast";
+import * as IpfsImage from './commands/ipfs-image';
+import { JSONContent } from "@tiptap/react";
+
+async function onPublishContent(blocks: JSONContent[], encryptionKey: CryptoKey | undefined) : Promise<JSONContent[]> {
```

Comment:
In a long and nested page, could the recursive functionality cause any problems? If there is, then I guess some error handling. If not, then not.

Reply:
Yes we could exceed the maximum call stack size if the recursion is too deep but there is no way to recover from it. So I think is better that we just let it fail and have it reported to Sentry.

We would need to apply a more sophisticated method if we exceed call stack. Right now I'm awaiting each level to mitigate to many parallel calls at least.